File: app/src/main/java/com/battlelancer/seriesguide/ui/people/PeopleActivity.java
package com.battlelancer.seriesguide.ui.people;

import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;
import androidx.annotation.Nullable;
import androidx.appcompat.app.ActionBar;
import butterknife.BindView;
import butterknife.ButterKnife;
import com.battlelancer.seriesguide.R;
import com.battlelancer.seriesguide.ui.BaseActivity;

public class PeopleActivity extends BaseActivity implements PeopleFragment.OnShowPersonListener {

    @Nullable
    @BindView(R.id.viewPeopleShadowStart)
    View shadowPeoplePane;
    @BindView(R.id.containerPeople) View containerPeople;

    private boolean isTwoPane;

    interface InitBundle {
        String MEDIA_TYPE = "media_title";
        String ITEM_TMDB_ID = "item_tmdb_id";
        String PEOPLE_TYPE = "people_type";
    }

    public enum MediaType {
        SHOW("SHOW"),
        MOVIE("MOVIE");

        private final String value;

        MediaType(String value) {
            this.value = value;
        }

        @Override
        public String toString() {
            return value;
        }
    }

    enum PeopleType {
        CAST("CAST"),
        CREW("CREW");

        private final String value;

        PeopleType(String value) {
            this.value = value;
        }

        @Override
        public String toString() {
            return value;
        }
    }

    static final int PEOPLE_LOADER_ID = 100;
    static final int PERSON_LOADER_ID = 101;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_people);
        ButterKnife.bind(this);
        setupActionBar();

        // if there is a pane shadow, we are in two pane layout
        isTwoPane = shadowPeoplePane != null;

        if (savedInstanceState == null) {
            // check if we should directly show a person
            int personTmdbId = getIntent().getIntExtra(PersonFragment.ARG_PERSON_TMDB_ID, -1);
            if (personTmdbId != -1) {
                showPerson(personTmdbId);

                // if this is not a dual pane layout, remove ourselves from back stack
                if (!isTwoPane) {
                    finish();
                    return;
                }
            }

            PeopleFragment f = new PeopleFragment();
            f.setArguments(getIntent().getExtras());

            // in two-pane mode, list items should be activated when touched
            if (isTwoPane) {
                f.setActivateOnItemClick();
            }

            getSupportFragmentManager().beginTransaction()
                    .add(R.id.containerPeople, f, "people-list")
                    .commit();
        }
    }

    @Override
    protected void setupActionBar() {
        super.setupActionBar();
        PeopleType peopleType = PeopleType.valueOf(
                getIntent().getStringExtra(InitBundle.PEOPLE_TYPE));
        setTitle(peopleType == PeopleType.CAST ? R.string.movie_cast : R.string.movie_crew);
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setHomeAsUpIndicator(R.drawable.ic_clear_24dp);
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setTitle(
                    peopleType == PeopleType.CAST ? R.string.movie_cast : R.string.movie_crew);
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int itemId = item.getItemId();
        if (itemId == android.R.id.home) {
            onBackPressed();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    public void showPerson(int tmdbId) {
        if (isTwoPane) {
            // show inline
            PersonFragment f = PersonFragment.newInstance(tmdbId);
            getSupportFragmentManager().beginTransaction()
                    .replace(R.id.containerPeoplePerson, f)
                    .commit();
        } else {
            // start new activity
            Intent i = new Intent(this, PersonActivity.class);
            i.putExtra(PersonFragment.ARG_PERSON_TMDB_ID, tmdbId);
            startActivity(i);
        }
    }
}

File: app/src/main/java/com/battlelancer/seriesguide/people/PeopleActivity.kt
package com.battlelancer.seriesguide.people

import android.content.Intent
import android.os.Bundle
import android.view.MenuItem
import android.view.ViewGroup
import com.battlelancer.seriesguide.R
import com.battlelancer.seriesguide.databinding.ActivityPeopleBinding
import com.battlelancer.seriesguide.people.PeopleFragment.OnShowPersonListener
import com.battlelancer.seriesguide.ui.BaseActivity
import com.battlelancer.seriesguide.util.ThemeUtils

/**
 * Displays a list of people, and on wide enough screens person details.
 * Otherwise lets [PersonActivity] show details.
 */
class PeopleActivity : BaseActivity(), OnShowPersonListener {

    private var isTwoPane = false

    internal interface InitBundle {
        companion object {
            const val MEDIA_TYPE = "media_title"
            const val ITEM_TMDB_ID = "item_tmdb_id"
            const val PEOPLE_TYPE = "people_type"
        }
    }

    enum class MediaType(private val value: String) {
        SHOW("SHOW"),
        MOVIE("MOVIE");

        override fun toString(): String = value
    }

    internal enum class PeopleType(private val value: String) {
        CAST("CAST"),
        CREW("CREW");

        override fun toString(): String = value
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = ActivityPeopleBinding.inflate(layoutInflater)
        setContentView(binding.root)
        ThemeUtils.configureForEdgeToEdge(binding.root as ViewGroup)
        setupActionBar()
        binding.sgAppBarLayout.sgAppBarLayout.liftOnScrollTargetViewId =
            PeopleFragment.liftOnScrollTargetViewId

        // If there is a details container, this is using a two pane layout.
        isTwoPane = binding.containerPeoplePerson != null

        val peopleFragment: PeopleFragment
        if (savedInstanceState == null) {
            // Check if this should directly show a person.
            val personTmdbId = intent.getIntExtra(PersonFragment.ARG_PERSON_TMDB_ID, -1)
            if (personTmdbId != -1) {
                showPerson(personTmdbId)

                // If this is not a dual pane layout, allow to go back directly by removing
                // this from back stack.
                if (!isTwoPane) {
                    finish()
                    return
                }
            }

            peopleFragment = PeopleFragment()
            peopleFragment.arguments = intent.extras
            supportFragmentManager.beginTransaction()
                .add(R.id.containerPeople, peopleFragment, "people-list")
                .commit()
        } else {
            peopleFragment =
                supportFragmentManager.findFragmentById(R.id.containerPeople) as PeopleFragment
        }
        if (isTwoPane) {
            // In two-pane mode, list items should be activated when touched.
            peopleFragment.setActivateOnItemClick()

            // In two-pane mode, details are displayed in a card that should
            // be adjusted for the nav bar.
            ThemeUtils.applyBottomMarginForNavigationBar(binding.cardPerson!!)
        }
    }

    override fun setupActionBar() {
        super.setupActionBar()
        val peopleType = PeopleType.valueOf(
            intent.getStringExtra(InitBundle.PEOPLE_TYPE)!!
        )
        setTitle(if (peopleType == PeopleType.CAST) R.string.movie_cast else R.string.movie_crew)
        val actionBar = supportActionBar
        if (actionBar != null) {
            actionBar.setHomeAsUpIndicator(R.drawable.ic_clear_24dp)
            actionBar.setDisplayHomeAsUpEnabled(true)
            actionBar.setTitle(
                if (peopleType == PeopleType.CAST) R.string.movie_cast else R.string.movie_crew
            )
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            android.R.id.home -> {
                onBackPressedDispatcher.onBackPressed()
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }

    override fun showPerson(tmdbId: Int) {
        if (isTwoPane) {
            // show inline
            val f = PersonFragment.newInstance(tmdbId)
            supportFragmentManager.beginTransaction()
                .replace(R.id.containerPeoplePerson, f)
                .commit()
        } else {
            // start new activity
            val i = Intent(this, PersonActivity::class.java)
            i.putExtra(PersonFragment.ARG_PERSON_TMDB_ID, tmdbId)
            startActivity(i)
        }
    }
}

interface PeopleActivityInterface {
    val isTwoPane: Boolean
}
